postgresql: add outputCheck for llvmPackages.llvm.lib

wolfgangwalther opened the pull request:
This should not leak into the out/lib outputs to keep the closure size small.

As discussed in https://github.com/NixOS/nixpkgs/pull/391950#discussion_r2029923376, @ma27.

I'm curious how many rebuilds we now get. I still have a few haskell-related postgresql->libpq changes open... but we should be a lot closer to ~500 now.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
postgresql: add outputCheck for llvmPackages.llvm.lib

This should not leak into the out/lib outputs to keep the closure size
small.

Ma27 commented:
> Can we avoid that somehow? We could for example not build any postgresql_jit or postgresqlJitPackages stuff on hydra. postgresql_jit and friends are only simple "link stuff together" derivations anyway and it would be enough to do that locally when you need them. WDYT, @Ma27?

The only thing on top is the buildEnv part to put the jit library back in, right? If so, I agree.

wolfgangwalther committed:
postgresqlJitPackages: move to aliases

Those are the same as postgresqlPackages now and we don't want them to
show up in CI's rebuild numbers.
